Make sure TARGET is set correctly when CURDIR contains spaces.

svendahlstrand opened the pull request:
With this fix the `TARGET` variable is set correctly when the project exists in
a path that contains spaces. So in this case:

    /Users/Joe/Dropbox (Personal)/example-project

`TARGET` will be set to `example-project` instead of `Dropbox example-project` (like it was
before this fix).

sej7278 commented:
Hi, thanks for the PR, i was hoping it would fix #372 but unfortunately it doesn't cover the use case where there are spaces in the ```TARGET``` directory, you end up with:

```
silly people/
└── use spaces
    └── in paths
        ├── blink.ino
        ├── build-leonardo
        │   ├── blink.ino.d
        │   ├── blink.ino.o
        │   ├── core
        │   │   ├── avr-libc
        │   │   │   ├── malloc.c.d
        │   │   │   ├── malloc.c.o
        │   │   │   ├── realloc.c.d
        │   │   │   └── realloc.c.o
        │   │   ├── CDC.cpp.d
        │   │   ├── CDC.cpp.o
        │   │   ├── ... blah ...
        │   │   └── WString.cpp.o
        │   ├── libcore.a
        │   ├── paths.eep
        │   ├── paths.elf
        │   ├── paths.hex
        │   └── paths.hex.sizeok
        └── Makefile
```

i.e. you get an elf file named after the ```lastword``` of the directory instead of the ino file, which may be way happened in #374 (if his directory was "something ATtiny").

Although as long as TARGET doesn't have a space, it seems ok - might be worth adding a note about that to #380.

svendahlstrand commented:
I try to avoid spaces in file and directory names myself, so I didn't think about the case when the base name contains spaces. You are absolutely right! :) I've fixed this pull request and updated #380 accordingly.

sej7278 commented:
could we make it replace spaces with underscores to be a bit more "standard" and so we don't get confused with the dash in ```OBJDIR```?

so we'd get something like:

```
space man/
├── blink.ino
├── build-uno/
│   ├── blink.ino.d
│   ├── blink.ino.o
│   ├── core/
│   ├── libcore.a
│   ├── space_man.eep
│   ├── space_man.elf
│   ├── space_man.hex
│   └── space_man.hex.sizeok
└── Makefile
```

it would need the text in #380 updating too :-(

then i'd say we can merge. you ok with that @sudar ?

svendahlstrand commented:
I've updated the Makefile and documentation to use underscore.

I'm not sure why the directory name is used as default `TARGET`. Is a `ino` file required for everything to work or could one use a `c` or `cpp` file instead? If so, maybe thats why the directory name is used.

sej7278 commented:
yes you can use c/cpp files not just ino, maybe that's why it uses the directory instead
